CMakeLists.txt

```diff
@@ -438,7 +448,7 @@ endif()
 
 set(USE_X11 0)
 
-if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT ENABLE_HEADLESS)
+if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT HAIKU AND NOT ENABLE_HEADLESS)
```

Comment:
Is X11 available on Haiku?
If so, why don't you want to use it there?
If not, why can't you just let the X11 detection fail? It's fine to do so and let it disable the X11 support then.